image/manifest.go

```diff
@@ -90,10 +88,11 @@ func manifestInstanceFromBlob(src types.ImageSource, manblob []byte, mt string)
 		return manifestSchema2FromManifest(src, manblob)
 	case manifest.DockerV2ListMediaType:
 		return manifestSchema2FromManifestList(src, manblob)
-	case "":
-		return nil, errors.New("could not guess manifest media type")
 	default:
-		return nil, fmt.Errorf("unsupported manifest media type %s", mt)
+		// if it's not a recognized manifest media type we'll try the last time
+		// to deserialize using v2s1 as per https://github.com/docker/distribution/blob/master/manifests.go#L108
+		// and https://github.com/docker/distribution/blob/master/manifest/schema1/manifest.go#L50
+		return manifestSchema1FromManifest(manblob)
```

Comment:
This doesn’t seem to be right. If, per https://bugzilla.redhat.com/show_bug.cgi?id=1389442, crane is sending essentially random MIME types depending on what the tag ends with, it will send these random MIME types also for schema2 manifests, or for any other future manifest format.